pkg/logs/agent_test.go

```diff
@@ -86,8 +86,8 @@ func createAgent(endpoints *config.Endpoints) (*Agent, *sources.LogSources, *ser
 }
 
 func (suite *AgentTestSuite) testAgent(endpoints *config.Endpoints) {
-	coreConfig.SetDetectedFeatures(coreConfig.FeatureMap{coreConfig.Docker: struct{}{}, coreConfig.Kubernetes: struct{}{}})
-	defer coreConfig.SetDetectedFeatures(coreConfig.FeatureMap{})
+	coreConfig.SetFeatures(coreConfig.Docker, coreConfig.Kubernetes)
```

Comment:
Since `SetFeatures` is used only in tests, would it make sense for it to take `t` as an argument and use `t.Cleanup` instead of relying on `defer`?

Reply:
Yes, great idea, implemented.